zipline/pipeline/loaders/blaze.py

```diff
@@ -0,0 +1,875 @@
+        Overwrites to the baseline data.
+    dates : pd.DatetimeIndex
+        The dates requested by the loader.
+
+    Returns
+    -------
+    new_baseline : pd.DataFrame
+        The new baseline data with novel deltas inserted.
+    """
+    get_indexes = dates.searchsorted
+    return pd.concat(
+        (baseline,
+         deltas.loc[
+             (get_indexes(deltas[TS_FIELD_NAME].values, 'right') -
+              get_indexes(deltas[AD_FIELD_NAME].values, 'left')) <= 1
+         ].drop(AD_FIELD_NAME, 1)),
+        ignore_index=True,
+    )
+
+
+def overwrite_from_dates(asof, dense_dates, sparse_dates, asset_idx, value):
+    """Construct a `Float64Overwrite` with the correct
+    start and end date based on the asof date of the delta,
+    the dense_dates, and the dense_dates.
+
+    Parameters
+    ----------
+    asof : datetime
+        The asof date of the delta.
+    dense_dates : pd.DatetimeIndex
+        The dates requested by the loader.
+    sparse_dates : pd.DatetimeIndex
+        The dates that appeared in the dataset.
+    asset_idx : int
+        The index of the asset in the block.
+    value : np.float64
+        The value to overwrite with.
+
+    Returns
+    -------
+    overwrite : Float64Overwrite
+        The overwrite that will apply the new value to the data.
+    """
+    first_row = dense_dates.searchsorted(asof)
+    last_row = dense_dates.get_loc(
+        sparse_dates[sparse_dates.searchsorted(asof) + 1],
+    ) - 1
+    if first_row > last_row:
+        return
+    yield Float64Overwrite(first_row, last_row, asset_idx, value)
+
+
+def adjustments_from_deltas_no_sids(dates,
+                                    dense_dates,
+                                    column_idx,
+                                    column_name,
+                                    assets,
+                                    deltas):
+    """Collect all the adjustments that occur in a dataset that does not
+    have a sid column.
+
+    Parameters
+    ----------
+    dates : pd.DatetimeIndex
+        The dates requested by the loader.
+    dense_dates : pd.DatetimeIndex
+        The dates that were in the dense data.
+    column_idx : int
+        The index of the column in the dataset.
+    column_name : str
+        The name of the column to compute deltas for.
+    deltas : pd.DataFrame
+        The overwrites that should be applied to the dataset.
+
+    Returns
+    -------
+    adjustments : dict[idx -> Float64Overwrite]
+        The adjustments dictionary to feed to the adjusted array.
+    """
+    ad_series = deltas.loc[:, AD_FIELD_NAME]
+    return {
+        dates.get_loc(kd): concat(tuple(
+            overwrite_from_dates(
+                ad_series.loc[kd],
+                dates,
+                dense_dates,
+                n,
+                v,
+            ) for n in range(len(assets)))
+        ) for kd, v in deltas[column_name].iteritems()
+    }
+
+
+def adjustments_from_deltas_with_sids(dates,
```

Comment:
Have we benchmarked the performance of this vs the DBZ?  Adjustments were a bottleneck for equity pricing data even when they were written in Cython.  I suspect this is going to be pretty hot code.